BaseListener/ApiGatewayFunction.cs

```diff
@@ -54,55 +64,46 @@ protected override void ConfigureServices(IServiceCollection services)
         /// <param name="context"></param>
         /// <returns></returns>
         [LogCall(LogLevel.Information)]
-        private async Task ProcessMessageAsync(SQSEvent.SQSMessage message, ILambdaContext context)
+        private async Task<APIGatewayProxyResponse> ProcessMessageAsync(APIGatewayProxyRequest apiGatewayProxyRequest, ILambdaContext context)
         {
-            context.Logger.LogLine($"Processing message {message.MessageId}");
+            context.Logger.LogLine($"Processing message {apiGatewayProxyRequest.RequestContext.MessageId}");
 
-            var entityEvent = JsonSerializer.Deserialize<EntityEventSns>(message.Body, _jsonOptions);
+            var eventType = apiGatewayProxyRequest.RequestContext.EventType;
 
-            using (Logger.BeginScope("CorrelationId: {CorrelationId}", entityEvent.CorrelationId))
+            try
             {
-                try
+                IDirectDebitProcessUseCase processor = null;
+                switch (eventType)
                 {
-                    IMessageProcessing processor = null;
-                    switch (entityEvent.EventType)
-                    {
-                        case EventTypes.DoSomethingEvent:
-                            {
-                                processor = ServiceProvider.GetService<IDoSomethingUseCase>();
-                                break;
-                            }
-                        // TODO - Implement other message types here...
-                        default:
-                            throw new ArgumentException($"Unknown event type: {entityEvent.EventType} on message id: {message.MessageId}");
-                    }
-
-                    await processor.ProcessMessageAsync(entityEvent).ConfigureAwait(false);
-                }
-                catch (Exception ex)
-                {
-                    Logger.LogError(ex, $"Exception processing message id: {message.MessageId}; type: {entityEvent.EventType}; entity id: {entityEvent.EntityId}");
-                    throw; // AWS will handle retry/moving to the dead letter queue
+                    case EventTypes.DirectDebitCalculationEvent:
+                        {
+                            processor = ServiceProvider.GetService<IDirectDebitProcessUseCase>();
+                            break;
+                        }
+                    default: throw new ArgumentException($"Unknown event type: {eventType} on message id: {apiGatewayProxyRequest.RequestContext.MessageId}");
                 }
+
+                return await processor.ProcessExecuteAsync(apiGatewayProxyRequest).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, $"Exception processing message id: {apiGatewayProxyRequest.RequestContext.MessageId}; type: {eventType}; request id: {apiGatewayProxyRequest.RequestContext.RequestId}");
+                throw; // AWS will handle retry/moving to the dead letter queue
```

Comment:
Might be best to confirm with Faisal about what even will trigger the lambda so we could know whether it will contain any data.

However, we shouldn't need any switch case because this is for the SQS template, where the SQS queue can have multiple types of events on it, which need to be processed differently.

This should also simplify the tests dramatically.

Reply:
Done